### PR TITLE
Add from_pretrained to dummy timm objects

### LINKED ISSUE #12091: Provide more useful error message in Detr from_pretrained when timm not installed
## Environment info

- `transformers` version: master
- Platform: Google Colab
- Python version: 3.7
- PyTorch version (GPU?):1.8.1+cu101
- Tensorflow version (GPU?):N/A
- Using GPU in script?:N/A
- Using distributed or parallel set-up in script?:N/A

## Information

Model I am using (Bert, XLNet ...):

The problem arises when using:
* [ ] the official example scripts: (give details below)
* [ ] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [ ] my own task or dataset: (give details below)

## To reproduce

[Here's a colab notebook.
](https://colab.research.google.com/drive/1b8RVCARcZU8kBFywRID8ZLPAYwhcd10p?usp=sharing)

1. install latest transformers from master w/o installing `timm`.
2. Try to init any `DetrModel` `from_pretrained`, and you'll see you get a misleading error
```python
from transformers import DetrForObjectDetection

model = DetrForObjectDetection.from_pretrained('facebook/detr-resnet-50')
```

Error thrown:

```bash
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-2-175d7bae5f8e> in <module>()
      1 from transformers import DetrForObjectDetection
      2 
----> 3 model = DetrForObjectDetection.from_pretrained('facebook/detr-resnet-50')

AttributeError: type object 'DetrForObjectDetection' has no attribute 'from_pretrained'
```

3. Try the same w/ timm installed, and see that it works.

## Expected behavior

Have message informing user that they must have `timm` installed to use `from_pretrained` with `Detr` models instead of just telling them there is no attribute `from_pretrained`.

### PR DESCRIPTION
**Pull request by LysandreJik**
Closes https://github.com/huggingface/transformers/issues/12091

cc @NielsRogge 

@sgugger Am I missing something relative to specifying that these dummy items should be generated with the `from_pretrained` method?

### CODE REVIEW

File: utils/check_dummies.py [hunk 1]
```diff
@@ -103,11 +103,14 @@ def read_init():
 def create_dummy_object(name, backend_name):
     """Create the code for the dummy object corresponding to `name`."""
     _pretrained = [
```
Comment (sgugger) [inline utils/check_dummies.py:105]: This is the list to modify to include new model patterns.

Comment (LysandreJik) [reply utils/check_dummies.py:105]: Aha, great!

File: utils/check_dummies.py [hunk 2]
```diff
@@ -103,11 +103,14 @@ def read_init():
...
-        "Config" "ForCausalLM",
+        "Config",
+        "ForCausalLM",
         "ForConditionalGeneration",
         "ForMaskedLM",
         "ForMultipleChoice",
+        "ForObjectDetection"
```
Comment (LysandreJik) [inline utils/check_dummies.py:111]:
```suggestion
        "ForObjectDetection",
```

**Comment by sgugger** [comment]
Failure of the templates came from the failure of `make quality` on master, fixed in a commit, so this is good to merge!

**Comment by LysandreJik** [comment]
Thanks a lot @sgugger!

**Pull request merged**